CMakeLists.txt

```diff
@@ -1,10 +1,65 @@
-cmake_minimum_required(VERSION 2.8)
+cmake_minimum_required(VERSION 3.0)
 project(termcolor)
 
-if(CMAKE_CXX_COMPILER_ID MATCHES "GNU|Clang")
-  set(CMAKE_CXX_FLAGS "-Werror -Wall -pedantic")
+# define target to install and link tests against
+add_library(termcolor INTERFACE)
```

Comment:
I think it is a matter of taste. Some want everything as variables, some want hard coded values. I was not sure how you wanted it, and it is very easy to change

the variable `CMAKE_PROJECT_NAME` is set with the `project(termcolor)` command. Therefore should work also in a submodule (and because of variable scopes should also not harm the parent project).